go/services/adr/gen.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+ROOT=$(git rev-parse --show-toplevel)
+
+"$(find "$ROOT/codegen/src" -name Azure.Iot.Operations.ProtocolCompiler -type f)" \
+	--modelFile "$ROOT/eng/dtdl/aep-name-based-operations.json" \
+	--outDir "$ROOT/go/services/adr/adrnamegen" \
```

Comment:
It looks like you're already re-exporting the types users will care about, so I'd probably put the generated files in `internal`.